examples/foomedical/src/pages/MessagesPage.tsx

```diff
@@ -1,122 +1,84 @@
+        return;
+      }
+
+      const practitionerRef = createReference(practitioner);
+
+      medplum
+        .createResource<Communication>({
+          resourceType: 'Communication',
+          status: 'in-progress',
+          sender: profileRef,
+          subject: profileRef,
+          recipient: [practitionerRef],
+          sent: new Date().toISOString(),
+          payload: [{ contentString: content }],
+        })
+        .catch(console.error);
+    },
+    [medplum, profileRef, practitioner]
+  );
+
+  if (!profileRef) {
+    return <Alert color="red">Error: Provider profile not found</Alert>;
+  }
 
-  if (!messages) {
+  if (!practitioner) {
     return <Loading />;
   }
 
   return (
     <Document width={800}>
-      <Title>Messages</Title>
-      <Divider my="xl" />
-      <Stack gap="xl">
-        {messages.map((resource) => (
-          <div key={resource.id}>
-            <Group align="top">
-              <ResourceAvatar size="lg" radius="xl" value={resource.sender?.resource as Practitioner} />
-              <div>
-                <Text size="sm" fw={500}>
-                  <ResourceName value={resource.sender?.resource as Patient | Practitioner} />
-                </Text>
-                <Text size="xs" color="dimmed">
-                  {formatDateTime(resource?.meta?.lastUpdated)}
-                </Text>
-                <Text size="md" my="sm">
-                  {resource.payload?.[0].contentString}
-                </Text>
-              </div>
-            </Group>
-          </div>
-        ))}
-        <div style={{ margin: '0 -20px -20px -20px', padding: 20, background: '#f8f8f8' }}>
-          <Form
-            onSubmit={(formData: Record<string, string>) => {
-              medplum
-                .createResource({
-                  resourceType: 'Communication',
-                  status: 'completed',
-                  subject: createReference(profile),
-                  sender: createReference(profile),
-                  payload: [{ contentString: formData.contentString }],
-                })
-                .then((result) => setMessages([...messages, result]))
-                .catch(console.log);
-            }}
-          >
-            <Group align="top">
-              <ResourceAvatar size="lg" radius="xl" value={profile} />
-              <Textarea name="contentString" style={{ flex: 1 }} placeholder="Add note" autosize />
-            </Group>
-            <Group justify="flex-end" mt="md">
-              <Button type="submit">Send</Button>
-            </Group>
-          </Form>
-        </div>
-      </Stack>
+      <BaseChat
+        title={`Chat with ${formatGivenName(practitioner.name?.[0] as HumanName)}`}
+        query={`sender=${getReferenceString(profile)},Practitioner/${practitionerId}&recipient=${getReferenceString(profile)},Practitioner/${practitionerId}`}
+        communications={communications}
+        setCommunications={(c) => setCommunications(c)}
```

Comment:
For some reason doing `setCommunications={setCommunications}` would only show new messages after a page refresh. LMK if I'm missing something here.

Reply:
@filipeximenes , I believe this is due to the fact that the `websockets` feature is not enabled on your project. Therefore, you were missing the realtime updates when your messages were saved. 

If this is true, you should be seeing errors in you JS console. If you slack me your project id, I can enable websockets on the project